lib/scan_result.dart

```diff
@@ -32,14 +44,31 @@ class ScanResult {
         advertisementData = AdvertisementData._fromJson(json);
 }
 
+/// A container for BLE advertisement data with convenient accessors for
+/// common items.
 class AdvertisementData {
+  /// The manufacturer data of the device.
   Uint8List manufacturerData;
+
+  /// A dictionary that contains service-specific advertisement data.
   Map<String, Uint8List> serviceData;
+
+  /// A list of service UUIDs.
   List<String> serviceUUIDs;
+
+  /// The local name of the device.
+  ///
+  /// It is usually the same as [Peripheral.name] but sometimes one of them or
```

Comment:
Not so sure about that.

Reply:
I checked it on many devices. I think it's worth to mention that there are two places where you can find the name of the device. So if one of them is null you can always try to use the other one, so you have better chance to display some name instead of _some device_.